lib/hawkular/inventory/inventory_api.rb

```diff
@@ -136,433 +131,122 @@ def get_config_data_for_resource(resource_path)
-        od = OperationDefinition.new ot
-        res.store od.name, od
+    def extract_entity_hash(parent_path, type, json, fetch_properties)
+      data = json['data']
+      type = shorten_resource_type(type)
+      data['path'] = "#{parent_path}/#{type};#{data['id']}"
+      if fetch_properties
+        props = fetch_properties(json)
+        data['properties'].merge! props unless props.nil?
       end
-      res
+      # Evict children
+      data.delete('children') unless data.key? 'children'
+      data
     end
 
-    # List operation definitions (types) for a given resource
-    # @param [String] resource_path canonical path of the resource entity
-    # @return [Array<String>] List of operation type ids
-    def list_operation_definitions_for_resource(resource_path)
-      resource = get_resource(resource_path.to_s, false)
-      list_operation_definitions(resource.type_path)
+    def fetch_properties(json)
+      return unless (json.key? 'children') && (json['children'].key? 'dataEntity')
+      config = json['children']['dataEntity'].find { |d| d['data']['id'] == 'configuration' }
+      config['data']['value'] unless config.nil?
     end
 
-    # Create a Metric and associate it with a resource.
-    # @param [String] metric_type_path Canonical path of the metric type of the new metric.
-    # @param [String] resource_path Canonical path of the resource to which we want to associate the metric.
-    # @param [String] metric_id Id of the metric
-    # @param [String] metric_name a (display) name for the metric. If nil, #metric_id is used.
-    # @return [Metric] The metric created or if it already existed the version from the server
-    def create_metric_for_resource(metric_type_path, resource_path, metric_id, metric_name = nil)
-      type_path = metric_type_path.is_a?(CanonicalPath) ? metric_type_path : CanonicalPath.parse(metric_type_path)
-      feed_id = type_path.feed_id
-      res_path = resource_path.is_a?(CanonicalPath) ? resource_path : CanonicalPath.parse(resource_path)
-      res_path_str = 'r;' + res_path.resource_ids.join('/r;')
-
-      m = {}
-      m['id'] = metric_id
-      m['name'] = metric_name || metric_id
-      m['metricTypePath'] = type_path.to_s
-
-      begin
-        http_post("/entity/f;#{feed_id}/metric", m)
-      rescue HawkularException => error
-        # 409 We already exist -> that is ok
-        raise unless error.status_code == 409
-      end
-
-      ret = http_get("/entity/f;#{feed_id}/m;#{metric_id}")
-      the_metric = Metric.new(ret)
-
-      begin
-        rl = {}
-        rl['otherEnd'] = the_metric.path.to_s
-        rl['name'] = 'incorporates'
-        http_post("/entity/f;#{feed_id}/#{res_path_str}/relationship", [rl])
-      rescue HawkularException => error
-        # 409 We already exist -> that is ok
-        raise unless error.status_code == 409
+    def shorten_resource_type(resource_type)
+      # cf InventoryStructure.class$EntityType
+      case resource_type
```

Comment:
yes, that would be pretty much the same... hash versus function, not sure what is the best